cropduster/static/cropduster/js/upload.js

```diff
@@ -127,6 +127,38 @@
         }
     };
 
+
+    var calcMinSize = function calcMinSize(size) {
+        var minSize = [size.min_w || size.w || 0, size.min_h || size.h || 0];
+        $.each(size.auto || [], function(i, autoSize) {
+            if (!autoSize.required) {
+                return;
+            }
+            var min_w = autoSize.min_w || autoSize.w || 0;
+            var min_h = autoSize.min_h || autoSize.h || 0;
+            minSize[0] = Math.max(minSize[0], min_w);
+            minSize[1] = Math.max(minSize[1], min_h);
+        });
+        return minSize;
+    };
+    var largest = function largest(array, initial, keyfunc) {
+        // Return largest element from array.
+        initial = initial || 0;
+        keyfunc = keyfunc || function identity(elm){ return elm; };
+        return [initial].concat(array).reduce(function take_larger(left, right){
+            return (keyfunc(left) >= keyfunc(right)) ? left: right;
+        });
+    }
+    // Fill in 'Min Size' help text
+    $(document).ready(function(){
+        var sizes = JSON.parse($('#id_crop-sizes').val());
+        var minSizes = sizes.map(calcMinSize);
+        var largest_width = largest(minSizes.map(function getWidth(sz){ return sz[0] }), 1);
+        var largest_height = largest(minSizes.map(function getHeight(sz){ return sz[1] }), 1);
+        $("#upload-help #min-width").html(String(largest_width));
```

Comment:
Does it? I think it's a wash. Either you have logic spread out to multiple places but the behavior is clearer when looking at one thing individually, or it's all contained in a single place. I come down on the side of keeping things together. Also the way you had it, it would show up as a dumb "Min. size: x" when it didn't apply (for instance, in CKEditor mode).